### DISCUSSION TITLE
How to pick ssl info from env variable

### DISCUSSION THREAD

**Discussion by villesau** [prisma/prisma#12546] [Q&A]
Hi, how can I pick the SSL information from an env variable?
Based on this: https://www.prisma.io/docs/concepts/database-connectors/postgresql#arguments it might not be possible.

However, this (if not possible) prevents from deploying to vercel & using postgres from google cloud. Vercel does not support mounting secret files, and google cloud understandably requires to use ssl certificates.

**Comment by austincrim** [comment]
Hey @villesau 👋 ,

What info specifically do you need? When do you need to read this information? At runtime or build time?

It might be possible to parse the connection string and search the query params for the SSL values, but I have never done something like that before.

**Reply by villesau** [reply]
Hi, I need sslcert & sslidentity at least. Vercel, for some reason, does not support having secret files so the only option is to have them as environment parameters. When connecting from public internet (in this case Vercel) to the sql instance, SSL encryption is pretty crucial.

**Reply by janpio** [reply]
Can you not for example store the values in a env var, then write that env var into a file on build of your serverless function and refer to that file in your connection string?

Besides that suggestion, please check if there is no feature request for this yet, and if there isn't, open one and describe your full use case. Thanks.

**Reply by villesau** [reply]
yeah everything is possible, it's software after all. But that does not mean it is very good or easy workflow if a lot of custom scaffolding is needed.

**Reply by janpio** [reply]
I am not saying that we will not consider a feature request because of that, I was just trying to propose a kind of straight forward workaround for you that you can use _now_.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
